gnd/src/main/java/com/google/android/gnd/persistence/local/room/FieldEntity.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.persistence.local.room;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Entity;
+import androidx.room.ForeignKey;
+import androidx.room.Index;
+import androidx.room.PrimaryKey;
+import com.google.android.gnd.model.form.Field;
+import com.google.android.gnd.model.form.Field.Type;
+import com.google.auto.value.AutoValue;
+import com.google.auto.value.AutoValue.CopyAnnotations;
+import java.util.List;
+
+@AutoValue
+@Entity(
+    tableName = "field",
+    foreignKeys =
+        @ForeignKey(
+            entity = FormEntity.class,
+            parentColumns = "id",
+            childColumns = "form_id",
+            onDelete = ForeignKey.CASCADE),
+    indices = {@Index("form_id")})
+public abstract class FieldEntity {
+
+  @CopyAnnotations
+  @NonNull
+  @PrimaryKey
+  @ColumnInfo(name = "id")
+  public abstract String getId();
+
+  @CopyAnnotations
+  @NonNull
+  @ColumnInfo(name = "type")
+  public abstract Type getType();
+
+  @CopyAnnotations
+  @Nullable
+  @ColumnInfo(name = "label")
+  public abstract String getLabel();
+
+  @CopyAnnotations
+  @ColumnInfo(name = "is_required")
+  public abstract boolean isRequired();
+
+  @CopyAnnotations
+  @Nullable
+  @ColumnInfo(name = "form_id")
+  public abstract String getFormId();
+
+  public static FieldEntity fromField(String formId, Field field) {
+    return FieldEntity.builder()
+        .setId(field.getId())
+        .setLabel(field.getLabel())
+        .setRequired(field.isRequired())
+        .setType(field.getType())
+        .setFormId(formId)
+        .build();
+  }
+
+  public static Field toField(FieldData fieldData) {
+    FieldEntity fieldEntity = fieldData.fieldEntity;
+    Field.Builder fieldBuilder =
+        Field.newBuilder()
+            .setId(fieldEntity.getId())
+            .setLabel(fieldEntity.getLabel())
+            .setRequired(fieldEntity.isRequired())
+            .setType(fieldEntity.getType());
+
+    List<MultipleChoiceEntity> multipleChoiceEntities = fieldData.multipleChoiceEntities;
+
+    // TODO: Re-enable this check.
+    //    if (multipleChoiceEntities.size() != 1) {
```

Comment:
We should recover gracefully on unexpected data in the db (Log.e, and ignore extra values for example), since this may happen with schema changes, etc.

Reply:
Done